Library/Homebrew/formula.rb

```diff
@@ -2495,6 +2496,7 @@ def cxxstdlib_check(check_type)
     # end</pre>
     def fails_with(compiler, &block)
       odisabled "fails_with :llvm" if compiler == :llvm
+      specs.each { |spec| spec.depends_on(SkipLinuxRequirement.new) } if compiler == :linux
```

Comment:
I had this way because something could e.g., `fails_with :gcc` and `fails_with :linux`. I can't seem to trigger that case though so I've made the change regardless.